### ISSUE TITLE
[Bug] multi-layer EAGLE DRAFT_EXTEND_V2 captures bs instead of num_tokens for attn-TP logprob metadata

### ISSUE THREAD

**Issue by glaziermag** [sgl-project/sglang#26416]
### Summary

`MultiLayerEagleDraftExtendCudaGraphRunner.get_forward_batch()` captures wrong DP/logprob metadata in the `require_attn_tp_gather` branch for `ForwardMode.DRAFT_EXTEND_V2`.

For the branch:

- `require_mlp_tp_gather == False`
- `require_attn_tp_gather == True`
- `forward_mode == DRAFT_EXTEND_V2`
- `speculative_num_steps >= 2`

current `main` initializes:

```python
global_num_tokens_for_logprob_gpu = [bs]
```

but DRAFT_EXTEND_V2 computes logits for all draft-extend tokens, so this should be `num_tokens = bs * self.num_tokens_per_bs`. The replay path already fills `bs * self.num_tokens_per_bs`, and the single-layer `eagle_draft_extend_cuda_graph_runner.py` uses `num_tokens_for_logprob = num_tokens` for DRAFT_EXTEND_V2.

Related open PR found before filing: #26016 appears to touch the broader `require_attn_tp_gather=True` path, but I did not find an open issue for this exact captured metadata mismatch.

### Environment

- SGLang source commit: `47617cc4df1e5c249d11ea5970ce26c32c8757af` (`main`)
- GPU: NVIDIA A100-SXM4-40GB
- Driver: 580.159.03
- CUDA runtime reported by PyTorch: 12.9
- PyTorch: 2.9.1+cu129
- GCP VM: `a2-highgpu-1g` Spot in `us-west4-b`
- Image: `pytorch-2-9-cu129-ubuntu-2204-nvidia-580-v20260526`

### Focused repro

I used a focused CUDA/PyTorch instrumentation script that imports and executes the real `MultiLayerEagleDraftExtendCudaGraphRunner.get_forward_batch()` implementation from the source tree, with unrelated SGLang modules stubbed out. This avoids needing a large multi-layer EAGLE model while still exercising the actual branch and CUDA tensor mutation.

Exact command on the A100 VM:

```bash
cd ~/sglang-repro
python3 scripts/repro_multi_layer_eagle_attn_tp_gather_metadata.py --bs 2 --speculative-num-steps 2
```

The script sets `step=1`, so `num_tokens_per_bs = speculative_num_steps + 1 + step = 4`, and `expected_num_tokens = bs * num_tokens_per_bs = 8`.

### Current main result

```json
{
  "branch_predicates": {
    "forward_mode": "DRAFT_EXTEND_V2",
    "require_attn_tp_gather": true,
    "require_mlp_tp_gather": false,
    "speculative_num_steps": 2
  },
  "bs": 2,
  "capture_metadata": {
    "forward_batch_global_dp_buffer_len": 8,
    "forward_batch_logits_rows": 8,
    "global_num_tokens_for_logprob_gpu": 2,
    "global_num_tokens_gpu": 8
  },
  "cuda_runtime": "12.9",
  "device": "NVIDIA A100-SXM4-40GB",
  "expected_num_tokens": 8,
  "num_tokens_per_bs": 4,
  "replay_metadata_fill": {
    "cuda_graph_readback_after_replay_fill": 8,
    "global_num_tokens_for_logprob_gpu": 8,
    "global_num_tokens_gpu": 8
  },
  "torch_version": "2.9.1+cu129",
  "wrong_capture_logprob_metadata": true
}
```

This shows capture initializes `global_num_tokens_for_logprob_gpu=2` (`bs`) while the expected DRAFT_EXTEND_V2 token count and replay metadata are both `8`.

### Patched result

After applying the one-line change below and rerunning the same command:

```json
{
  "capture_metadata": {
    "forward_batch_global_dp_buffer_len": 8,
    "forward_batch_logits_rows": 8,
    "global_num_tokens_for_logprob_gpu": 8,
    "global_num_tokens_gpu": 8
  },
  "expected_num_tokens": 8,
  "replay_metadata_fill": {
    "cuda_graph_readback_after_replay_fill": 8,
    "global_num_tokens_for_logprob_gpu": 8,
    "global_num_tokens_gpu": 8
  },
  "wrong_capture_logprob_metadata": false
}
```

### Minimal diff

```diff
diff --git a/python/sglang/srt/speculative/multi_layer_eagle_draft_extend_cuda_graph_runner.py b/python/sglang/srt/speculative/multi_layer_eagle_draft_extend_cuda_graph_runner.py
@@
             buffers.global_num_tokens_for_logprob_gpu.copy_(
                 torch.tensor(
-                    [bs],
+                    [num_tokens],
                     dtype=torch.int32,
                     device=buffers.input_ids.device,
                 )
```

### Expected behavior

For DRAFT_EXTEND_V2 in the attention-TP gather branch, capture should initialize `global_num_tokens_for_logprob_gpu` with the number of logits/logprob hidden states (`num_tokens`), matching:

- `next_token_logits_buffer[:num_tokens]`
- `global_dp_buffer_len = num_tokens`
- replay's `buffers.global_num_tokens_for_logprob_gpu.fill_(bs * self.num_tokens_per_bs)`
- single-layer EAGLE draft-extend CUDA graph runner behavior
